core/logger/logger.go

```diff
@@ -85,10 +133,47 @@ func CreateProductionLogger(
 	config.Level.SetLevel(lvl)
 
 	zl, err := config.Build(zap.AddCallerSkip(1))
+
 	if err != nil {
 		log.Fatal(err)
 	}
-	return &Logger{
-		SugaredLogger: zl.Sugar(),
+	return CreateLoggerWithConfig(zl.Sugar(), dir, jsonConsole, toDisk)
+}
+
+func (l *Logger) InitServiceLevelLogger(serviceName string, logLevel string) (*Logger, error) {
+	config := zap.NewProductionConfig()
+	if !l.jsonConsole {
+		config.OutputPaths = []string{"pretty://console"}
+	}
+	if l.toDisk {
+		destination := logFileURI(l.dir)
+		config.OutputPaths = append(config.OutputPaths, destination)
+		config.ErrorOutputPaths = append(config.ErrorOutputPaths, destination)
+	}
```

Comment:
Can we abstract this stuff?

Reply:
makes sense, can abstract the config curation out, something like?: 

`config := initLogConfig(dir, jsonConsole, lvl, toDisk)
 config.Build(...)
`